patches/cli/0012-allow-intermediate-package-cache.patch

```diff
@@ -0,0 +1,52 @@
+From d8d39ebf8c98a5339089cb89d5838e4ece71780f Mon Sep 17 00:00:00 2001
+From: karajas <karajas@microsoft.com>
+Date: Wed, 6 Sep 2017 18:33:13 -0400
+Subject: [PATCH] Allow cli to restore to an intermdiate package cache to pick up source built packages 
+
+---
+ build/Prepare.targets | 22 +++++++++++++++++++++-
+ 1 file changed, 21 insertions(+), 1 deletion(-)
+
+diff --git a/build/Prepare.targets b/build/Prepare.targets
+index 214b1f7..b94b60a 100644
+--- a/build/Prepare.targets
++++ b/build/Prepare.targets
+@@ -83,6 +83,26 @@
+                    ProjectPath="&quot;%(RestoreToolsPackagesInput.FullPath)&quot;"
+                    AdditionalParameters="/p:UsePortableLinuxSharedFramework=$(UsePortableLinuxSharedFramework)" />
+ 
++
++      <PropertyGroup>
++        <IntermediateToolsPackages>$(MSBuildThisFileDirectory)packages</IntermediateToolsPackages>
++        <FilesToCopyPattern>$(MSBuildThisFileDirectory)packages/**/*</FilesToCopyPattern>
++      </PropertyGroup>
++  
++      <DotNetRestore ToolPath="$(Stage0Directory)"
```

Comment:
This is a pretty harsh hack, and it doesn't address the underlying problem.

The real problem here is that during `source-build`, we are getting NuGet cache polluted with packages that weren't built during `source-build`.  For example, Microsoft already built and shipped `Microsoft.NETCore.App v2.0.0`.  But when building `source-build`, it ALSO produces `Microsoft.NETCore.App v2.0.0`.  These two packages have the same identity, but they have different contents in them.

During the build of `source-build`, the Microsoft shipped package of `Microsoft.NETCore.App v2.0.0` is coming from nuget.org and getting put in the NuGet package cache.  Later, when we try to use the package we built, it sees it already has `Microsoft.NETCore.App v2.0.0` in the cache and doesn't use the one built from the repo.

We need to ensure the build is always using the package that just got built.  We will get the wrong assets in places if we don't - for example, for the native built assets we can get the linux-x64 portable version instead of the distro-specific version that was just built.

Reply:
Agree, this was a quick workaround NOT the final solution, I wanted to unblock folks who were looking into this. The tough part is the versioning being the same for both the package being built in source-build and the one being produced officially. I'll open up an issue about this. Do you have any suggestions moving forward to address this conflict?

Reply:
One suggestion would be that we use different package cache locations for "tools" and for "product" packages.

ex. When restoring BuildTools, or any other NuGet packages that come from external sources, use a directory named `externalPackages`.  When restoring actual product projects that depend on packages that were built earlier in the source-build process, use a directory named `productPackages`.

The names can be anything, as long as they are different.  But the idea being split the package cache for packages that are "tooling/bootstrapping" and packages that are used in the end product.